### ISSUE TITLE
Incorrect evaluation of "keep-alive-enabled" parameter in RESTEasy Reactive client

### ISSUE THREAD

**Issue by gefloh** [quarkusio/quarkus#31480]
### Describe the bug

The parameter `keep-alive-enabled` that was added in #31125  is currently evaluated incorrectly because a default value of `true` is set not only for the global configuration, but also for the class name/config key values. This leads e.g. to a global value of `false` being overwritten by the default value of `true` from the classname config which is not the expected behavior.

Evaluation:
```
Optional<Boolean> keepAliveEnabled = oneOf(clientConfigByClassName().keepAliveEnabled,
                clientConfigByConfigKey().keepAliveEnabled, configRoot.keepAliveEnabled);
```

resolves always to the `clientConfigByClassName` value if set (which is always the case because of the default value).

I will fix this directly myself by removing the default value for `keepAliveEnabled` in `io.quarkus.restclient.config.RestClientConfig`.

### Expected behavior

Setting the global configuration to `false` for `keep-alive-enabled` without any other config value provided should disable the keep alive (same for config key).

### Actual behavior

The global configuration is overwritten by the incorrect default value from `clientConfigByClassName` (same for config key).

**Comment by gefloh** [comment]
Will be fixed with #31482.

**Comment by TomasHofman** [comment]
Right, this wasn't obvious. I agree with the solution. Thanks for pinging me :).

**Comment by gefloh** [comment]
Yes, true. Unfortunately I am still not able to test these things with my local built Quarkus version. I am not sure if the contributing guide is correct/up-to-date.

**Comment by geoand** [comment]
What kind of problems are you having?

**Comment by gefloh** [comment]
I did a `./mvnw -Dquickly`, this worked fine. The `quarkus.platform.group-id` property in the `pom.xml` was already set to `io.quarkus`, so I changed only `quarkus.platform.version` to `999-SNAPSHOT`. 

`mvn clean compile` gives me several compilation errors like `package javax.inject does not exist`. This was a freshly created Quarkus Lambda, so nothing changed there from my side. Maybe I'm missing something?
